saga/sagalogs/memory.go

```diff
@@ -0,0 +1,155 @@
+// Package sagalogs provides implementations of SagaLog.
+package sagalogs
+
+import (
+	"errors"
+	"fmt"
+	"github.com/twitter/scoot/saga"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// In Memory Implementation of a SagaLog, DOES NOT durably persist messages.
+type inMemorySagaLog struct {
+	sagas        map[string]*logData
+	mutex        sync.RWMutex
+	gcExpiration time.Duration
+	gcTicker     *time.Ticker
+}
+
+// wrapper for SagaMessages tracking timestamps for GC
+type logData struct {
+	messages []saga.SagaMessage
+	created  time.Time
+}
+
+// Returns an Instance of a Saga based on an inMemorySagaLog.
+// This is an in-memory impl, and is not durable.
+// Implements GC of whole sagas based on time expiration.
+// GC should be set at a duration that realistically will not purge active Sagas.
+// gcExpiration: duration after which a Saga was created, it will be deleted.
+// 	A zero duration is interpretted as "never gc" (the Log will eventually consume all memory).
+// gcInterval: duration interval at which GC runs.
+func MakeInMemorySagaCoordinator(gcExpiration time.Duration, gcInterval time.Duration) saga.SagaCoordinator {
```

Comment:
can we have units on gcInterval variable name: gcInterval_sec?

Reply:
time.Duration type makes that moot - this stops being a "sec" field after reading it in from the config